### PR TITLE
Modernise the library

### PR DESCRIPTION
**Pull request by paskal**
Bump  golangci-lint to v1.47.2, get rid of github.com/pkg/errors in favour of stdlib, bump go modules.

### CODE REVIEW

File: cache_test.go [hunk 1]
```diff
@@ -1,16 +1,15 @@
 package lcw
 
 import (
-	"errors"
 	"fmt"
+	"github.com/go-redis/redis/v8"
```
Comment (umputun) [inline cache_test.go]: nit: this import should not be inside stdlib imports

File: go.mod [hunk 2]
```diff
@@ -3,11 +3,12 @@ module github.com/go-pkgz/lcw
 go 1.15
```
Comment (umputun) [inline go.mod:3]: ci switched to 1.18, but this one still 1.15. I suggest bumping to 1.18 as well

File: lru_cache.go [hunk 3]
```diff
@@ -1,13 +1,12 @@
 package lcw
 
 import (
+	"fmt"
 	"sync/atomic"
 
+	"github.com/go-pkgz/lcw/eventbus"
```
Comment (umputun) [inline lru_cache.go:7]: nit: you IDE has some odd import sorting settings

**Review by umputun — CHANGES_REQUESTED** [review]

**Review by umputun — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (paskal) [reply go.mod:3]: I'll change this one alongside the generics introduction. Now the minimum because of `%w, err` formatting is 1.13.

**Pull request merged**